examples/pub-from-chrome/sfu.js

```diff
@@ -0,0 +1,207 @@
+  }
+})
+
+
+
+
+
+const join = async () => {
+    const offer = await pcPublish.createOffer()
+    await pcPublish.setLocalDescription(offer)
+    const id = uuidv4()
+
+    log("Sending join")
+    log(pcPublish.localDescription)
+    socket.send(JSON.stringify({
+        method: "join",
+        params: { sid: "test room", offer: pcPublish.localDescription },
+        id
+    }))
+
+
+    socket.addEventListener('message', (event) => {
+        const resp = JSON.parse(event.data)
+        if (resp.id === id) {
+            log(`Received publish answer`)
+
+            // Hook this here so it's not called before joining
+            pcPublish.onnegotiationneeded = async function () {
+                log("onnegotiationneeded")
+                const offer = await pcPublish.createOffer()
+                await pcPublish.setLocalDescription(offer)
+                const id = uuid.v4()
+                socket.send(JSON.stringify({
+                    method: "offer",
+                    params: { desc: offer },
+                    id
+                }))
+
+                socket.addEventListener('message', (event) => {
+                    const resp = JSON.parse(event.data)
+                    if (resp.id === id) {
+                        log(`Got renegotiation answer`)
+                        pcPublish.setRemoteDescription(resp.result)
+                    }
+                })
+            }
+
+            log(`setRemoteDescription`)
+            // log(resp)
+            pcPublish.setRemoteDescription(resp.result)
+        } else if (resp.method == "trickle"){
+            log("receive trickle")
+            // log(resp.params)
+            // const candidate: RTCIceCandidateInit = "a="+resp.params.candidate
+            pcPublish.addIceCandidate(resp.params)
+        }
+    })
+}
+
+
+// click pub button
+window.Pub = () => {
+    log("Publishing stream")
+
+    localStream.getTracks().forEach((track) => {
+        pcPublish.addTrack(track, localStream);
+    });
+
+
+    join()
+}
+
+
+// window.Sub = name => {
+//     let pcSubcribe = new RTCPeerConnection({
+//         // iceServers: [{
+//         //     urls: 'stun:stun.l.google.com:19302'
+//         // }]
+//     })
+
+
+//     // 1. send subscribe  sdp
+//     pcSubcribe.onicecandidate = event => {
+//         if (event.candidate === null) {
+//             log("SDP chrome ->  sfu:\n" + pcSubcribe.localDescription.sdp)
+//             var sendData = {type:'subscribe', sdp:pcSubcribe.localDescription.sdp, name:name}
+//             sock.send(JSON.stringify(sendData));
+//         }
+//     }
+
+//     pcSubcribe.addTransceiver('audio', {'direction': 'recvonly'})
+//     pcSubcribe.addTransceiver('video', {'direction': 'recvonly'})
+
+//     pcSubcribe.createOffer()
+//         .then(d => pcSubcribe.setLocalDescription(d))
+//         .catch(log)
+
+//     // 4. receive data
+//     pcSubcribe.ontrack = function (event) {
+//         var el = document.getElementById('remote')
```

Comment:
The `remote` HTML element is currently a div, but you use it like it is a video element. I think you meant to create a video tag inside the div and display the video using this tag.

Reply:
done . delete unused code.